dragon-scales/src/app/components/StepViewerComponents/StepViewer.jsx

```diff
@@ -169,108 +224,87 @@ export default function StepViewer( { stepEntity, markStepState, reloadTask } )
             console.error("Error deleting step:", error);
         }
     };
-    useEffect(() => {
-        const handleClickOutside = (event) => {
-            if (stepViewerRef.current && !stepViewerRef.current.contains(event.target)) {
-                deactivateStepViewer();
-            }
-        };
 
-        if (isActive) {
-            document.addEventListener("click", handleClickOutside);
-        }
-        return () => {
-            document.removeEventListener("click", handleClickOutside);
-        }
-    }, [isActive]);
+    const handleOpenEditDialog = (event) => {
+        event.stopPropagation();
+        setEditDialogOpen(true);
+    };
+
+    const handleCloseEditDialog = () => {
+        setEditDialogOpen(false);
+    };
 
+    // Handle successful edit
+    const handleEditSuccess = (updatedStep) => {
+        setStep(updatedStep);
+    };
```

Comment:
What is this doing? If I understand this correctly, on success you are just triggering the reload function right? so why are you changing step here and what is updatedStep? I can't get a log statement on my browser when changing the step name